lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -107,8 +107,10 @@ defmodule Kernel.ParallelCompiler do
         end)
       end
 
-    spawn_compilers(t, original, output, options, waiting,
-                    [{pid, ref, h} | queued], schedulers, result)
+    timer_ref = Process.send_after(self(), {:timed_out, pid}, options[:waiting_timeout] || 5_000)
```

Comment:
I think this line is triggering the build failure as you are using the access protocol. Those are the famous bootstrap issues. Many things, like the compiler, require us to use a subset of Elixir.